src/main/kotlin/org/sd_jwt/SdJwt.kt

```diff
@@ -210,6 +244,72 @@ inline fun <reified T> createCredential(
     return sdJwtEncoded + SEPARATOR + disclosures.joinToString(SEPARATOR)
 }
 
+/**
+ * @suppress
+ * This method is not for API users.
+ *
+ * Verifies if keys (and sub keys) of firstJson exist in secondJson
+ *
+ */
+fun validateJSON(
+    firstJson: JSONObject,
+    secondJson: JSONObject
+): Boolean {
+    val keys = firstJson.keys()
+
+    while (keys.hasNext()) {
+        val key = keys.next() as String
+
+        if (secondJson.has(key).not()) {
+            return false
+        }
+
+        val value1 = firstJson[key]
+        val value2 = secondJson[key]
+
+        // Recursively check the structure for nested JSONObjects or JSONArrays
+        if (value1 is JSONObject && value2 is JSONObject) {
+            if (!validateJSON(value1, value2)) {
+                return false
+            }
+        } else if (value1 is JSONArray && value2 is JSONArray) {
+            if (!validateJSONArray(value1, value2)) {
+                return false
+            }
+        } else if (
+            value1 is JSONArray && value2 !is JSONArray ||
+            value1 !is JSONArray && value2 is JSONArray
+        ) {
+            // value types should match
+            return false
+        }
+    }
+
+    return true
+}
+
+/**
+ * @suppress
+ * This method is not for API users.
+ *
+ * Verifies if elements of firstArray and secondArray are matching
+ *
+ */
+fun validateJSONArray(firstArray: JSONArray, secondArray: JSONArray): Boolean {
+    for (i in 0 until firstArray.length()) {
+        if (firstArray[i] is JSONObject && secondArray[i] is JSONObject) {
+            if (!validateJSON(firstArray.getJSONObject(i), secondArray.getJSONObject(i))) {
+                return false
+            }
+        } else if (firstArray[i] is JSONArray && secondArray[i] is JSONArray) {
+            if (!validateJSONArray(firstArray.getJSONArray(i), secondArray.getJSONArray(i))) {
+                return false
+            }
+        }
```

Comment:
Shouldn't there be an else branch returning false because in case there is a JSON array in the first array and a JSON object in the second array and vice versa the function should return false, right?

Reply:
very good catch! 
I have adjusted this function to cover that case, and added further test cases for that.